### ISSUE TITLE
Pretty-print actual output when HTML comparison fails, for more precise diff

### ISSUE THREAD

**Issue by mattbrictson** [bootstrap-ruby/bootstrap_form#399]
When a test using `assert_equivalent_xml` fails, the expected HTML is nicely indented, but the actual HTML is not. This results in a diff that is hard to interpret, as shown here:

```diff
-<div class="form-group row">
-  <div class="col-sm-10 col-sm-offset-2">Hallo</div>
-</div>
-<div class="form-group row">
-  <label class="col-sm-2 required" for="user_email">Email</label>
-  <div class="col-sm-10">
-    <input class="form-control" id="user_email" name="user[email]" type="text" value="steve@example.com"/>
-  </div>
-</div>
-
+<div class="form-group row"><div class="col-sm-10 col-sm-offset-2">Hallo</div></div><div class="form-group row"><label class="col-form-label col-sm-2 required" for="user_email">Email</label><div class="col-sm-10"><input class="form-control" id="user_email" name="user[email]" type="text" value="steve@example.com"/></div></div>
```

It would be great if _both_ the actual and expected output were nicely indented, so that a more precise diff could be printed, like this:

```diff
   <div class="col-sm-10 col-sm-offset-2">Hallo</div>
 </div>
 <div class="form-group row">
-  <label class="col-sm-2 required" for="user_email">Email</label>
+  <label class="col-form-label col-sm-2 required" for="user_email">Email</label>
   <div class="col-sm-10">
     <input class="form-control" id="user_email" name="user[email]" type="text" value="steve@example.com"/>
   </div>
```

(This may or not be possible using the tools available to us via the `equivalent-xml` gem; I'm not sure.)

**Comment by mattbrictson** [comment]
This seems to do the trick. I'll do more thorough testing and open a PR later this week, unless someone else wants to jump on it first.

```diff
diff --git a/test/test_helper.rb b/test/test_helper.rb
index daf29da..b677f2d 100644
--- a/test/test_helper.rb
+++ b/test/test_helper.rb
@@ -53,8 +53,8 @@ class ActionView::TestCase
 
   # Expected and actual are wrapped in a root tag to ensure proper XML structure
   def assert_equivalent_xml(expected, actual)
-    expected_xml        = Nokogiri::XML("<test-xml>\n#{expected}\n</test-xml>")
-    actual_xml          = Nokogiri::XML("<test-xml>\n#{actual}\n</test-xml>")
+    expected_xml        = Nokogiri::XML("<test-xml>\n#{expected}\n</test-xml>") { |config| config.default_xml.noblanks }
+    actual_xml          = Nokogiri::XML("<test-xml>\n#{actual}\n</test-xml>") { |config| config.default_xml.noblanks }
     ignored_attributes  = %w(style data-disable-with)
 
     equivalent = EquivalentXml.equivalent?(expected_xml, actual_xml, {
@@ -83,8 +83,8 @@ class ActionView::TestCase
     assert equivalent, lambda {
       # using a lambda because diffing is expensive
       Diffy::Diff.new(
-        sort_attributes(expected_xml.root),
-        sort_attributes(actual_xml.root)
+        sort_attributes(expected_xml.root).to_xml(indent: 2),
+        sort_attributes(actual_xml.root).to_xml(indent: 2)
       ).to_s(:color)
     }
   end
```

**Comment by lcreid** [comment]
This would help what I'm doing now, so I'm going to take a quick run at it.

**Comment by mattbrictson** [comment]
Closed via #404
